wrappers/src/filter.hpp

```diff
@@ -46,12 +46,45 @@ inline Results* get_filtered_results(const SharedRealm& realm, const ConstTableR
 
     std::vector<Mixed> mixed_args;
     mixed_args.reserve(args_count);
+
+    std::vector<Geospatial> geo_store;
+
+    int geo_store_size = 0;
     for (size_t i = 0; i < args_count; ++i) {
-        if (arguments[i].type != realm_value_type::RLM_TYPE_LINK) {
-            mixed_args.push_back(from_capi(arguments[i]));
+        switch (arguments[i].type) {
+        case query_argument_type::BOX:
+        case query_argument_type::POLYGON:
+        case query_argument_type::SPHERE:
+            ++geo_store_size;
+            break;
         }
-        else {
-            mixed_args.push_back(from_capi(arguments[i].link.object, true));
+    }
+
+    geo_store.reserve(geo_store_size);
```

Comment:
Consider allocating `args_count` for the geo_store, even though they're not likely to be all geo - it simplifies the code.